concent_api/core/tests/test_auth_accept_or_reject_results.py

```diff
@@ -0,0 +1,1110 @@
+                    HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = self._get_encoded_requestor_public_key(),
+                )
+
+        self._test_response(
+            response,
+            status       = 200,
+            key          = self.PROVIDER_PRIVATE_KEY,
+            message_type = message.concents.ServiceRefused,
+            fields       = {
+                'reason':    message.concents.ServiceRefused.REASON.DuplicateRequest,
+                'timestamp': self._parse_iso_date_to_timestamp("2018-02-05 10:00:31"),
+            }
+        )
+        self._test_auth_message_counter_not_increased()
+
+    def test_requestor_should_receive_subtask_results_from_concent_with_correct_keys(self):
+        """
+        Test if Provider submitted ForceSubtaskResults, Concent will return message to Requestor with new timestamp
+        if Requestor ask Concent before deadline if correct keys are used.
+
+        Expected message exchange:
+        Provider    -> Concent:                   ForceSubtaskResults
+        Concent     -> Provider:                  HTTP 202
+        Concent     -> WrongRequestor/Provider:   HTTP 204
+        Concent     -> Requestor:                 ForceSubtaskResults (new timestamp)
+        """
+
+        # STEP 1: Provider forces subtask results via Concent.
+        # Request is processed correctly.
+        serialized_force_subtask_results = self._get_serialized_force_subtask_results(
+            timestamp                   = "2018-02-05 10:00:20",
+            ack_report_computed_task    = self._get_deserialized_ack_report_computed_task(
+                timestamp       = "2018-02-05 10:00:20",
+                subtask_id      = "xxyyzz",
+                task_to_compute = self._get_deserialized_task_to_compute(
+                    timestamp   = "2018-02-05 10:00:00",
+                    deadline    = "2018-02-05 10:00:10",
+                    task_id     = '2',
+                )
+            )
+        )
+
+        with mock.patch('core.views.is_provider_account_status_positive', _get_provider_account_status_true_mock):
+            with freeze_time("2018-02-05 10:00:31"):
+                response = self.client.post(
+                    reverse('core:send'),
+                    data                                = serialized_force_subtask_results,
+                    content_type                        = 'application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY      = self._get_encoded_provider_public_key(),
+                    HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = self._get_encoded_requestor_public_key(),
+                )
+
+        assert len(response.content) == 0
+        assert response.status_code  == 202
+
+        self._test_database_objects(
+            last_object_type         = message.concents.ForceSubtaskResults,
+            task_id                  = '2',
+            receive_delivered_status = False,
+        )
+        self._test_auth_message_counter_increased()
```

Comment:
These methods look like test assertions. Their names should start with `assert_` like they do for built-in assertions.

Reply:
:+1: